src/lexer.c

```diff
@@ -250,29 +250,65 @@ void tryParseImportStatement () {
       if (ch == 'm' && str_eq3(pos + 1, 'e', 't', 'a') && *lastTokenPos != '.')
         addImport(startPos, startPos, pos + 4, IMPORT_META);
       return;
-    
+
     default:
       // no space after "import" -> not an import keyword
       if (pos == startPos + 6)
         break;
     case '"':
     case '\'':
-    case '{':
-    case '*':
+    case '*': {
       // import statement only permitted at base-level
       if (openTokenDepth != 0) {
         pos--;
         return;
       }
       while (pos < end) {
         ch = *pos;
-        if (ch == '\'' || ch == '"') {
+        if (isQuote(ch)) {
           readImportString(startPos, ch);
           return;
         }
         pos++;
       }
       syntaxError();
+      break;
+    }
+
+    case '{': {
+      // import statement only permitted at base-level
+      if (openTokenDepth != 0) {
+        pos--;
+        return;
+      }
+      char16_t firstQuote;
+      bool curlyBraceClosed = false;
+      bool quoteClosed = true;
+      
+      while (pos < end) {
+        ch = *pos;
+        if (isQuote(ch)) {
+          if (curlyBraceClosed) {
+            readImportString(startPos, ch);
+            return;
+          }
+
+          if (!firstQuote) {
+            firstQuote = ch;
+            quoteClosed = false;
+          } else if (ch == firstQuote) {
+            quoteClosed = true;
+          }
+        } else if (ch == '}') {
+          curlyBraceClosed = quoteClosed;
```

Comment:
Rather than handling this as a variable and keeping the loop going, just break the loop and have the `readImportString` check after this loop.